src/include/btree_inline.h

```diff
@@ -2269,6 +2269,28 @@ __wt_btcur_bounds_early_exit(
     return (0);
 }
 
+/*
+ * __wt_cbt_set_valid_data_flag --
+ *     Sets the page valid data flag if it has any data other than deleted.
+ */
+static inline void
+__wt_cbt_set_valid_data_flag(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *first_upd)
+{
+    if (cbt->upd_value->tw.stop_txn != WT_TXN_MAX) {
+        if (__wt_txn_upd_value_visible_all(session, cbt->upd_value))
+            ++cbt->page_obsolete_deleted_count;
+        /*
+         * If the selected tombstone is not first in the update list indicates that there are newer
+         * updates in the list that is either not committed or not visible.
+         */
+        else if (!cbt->valid_data && first_upd &&
+          (cbt->upd_value->tw.stop_txn != first_upd->txnid ||
```

Comment:
I don't understand why we compare the selected update's timestamp window against the latest update's starts. If for instance "cbt->upd_value->tw.stop_txn == first_upd->txnid" what does that mean? that the transaction that did the latest update also was the one that deleted the selected update? and in such a case we do not have valid data on the page?

Reply:
This check to avoid unnecessary eviction of a page when it doesn't find any newer updates of a key due to an older snapshot. This is identified by comparing what is tombstone txnid that it can see compared to what is the most recent update in the update list indicates that there are further updates on the update list, so it is better not to evict the page.